src/availability.rs

```diff
@@ -579,6 +656,38 @@ mod test {
                         .hash()
                 );
             }
+
+            let block_range: Vec<BlockQueryData<MockTypes>> = client
```

Comment:
Should I add tests to compare whole range or is that not neccessary?

Reply:
Compare against what? Fetching each block in the range individually? It would be good but would slow the test down a lot, and we already have tasks for the underlying `get_*_range` functions in `data_source.rs` (`test_range`). Maybe just spot check the most recent block in each range?

Reply:
I was thinking of collecting the individual blocks already fetched and comparing them, but since we are not doing that for all the blocks, this approach would fail.